### PR TITLE
Enhancing documentation for viral-pipelines

### PR DESCRIPTION
**Pull request by golu099**
Adding documentation to important input/ouput parameters for some of the more used workflows in the pipeline (assembled_refbased, demux_deplete,classify_single, augur_from_assemblies, and genbank.

Additionally orphan workflows will be noted.

### CODE REVIEW

File: pipes/WDL/workflows/16S_train_classifier.wdl [hunk 1]
```diff
@@ -0,0 +1,26 @@
+version 1.0
```
Comment (dpark01) [inline pipes/WDL/workflows/16S_train_classifier.wdl:1]: delete this wdl file from this branch

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (dpark01) [inline .vscode/settings.json:1]: delete this file from the branch and perhaps add `.vscode/` to the `.gitignore` file instead?

**Review by dpark01 — CHANGES_REQUESTED** [review]
A few housekeeping requests -- haven't dug into content yet

File: docs/workflows.rst [hunk 3]
```diff
@@ -3,8 +3,9 @@ WDL Workflows
 
 Documentation for each workflow is provided here. Although there are many workflows that serve different functions, some of the primary workflows we use most often include:
 
- - :doc:`demux_plus` (on every sequencing run)
- - :doc:`classify_multi` (metagenomics & host depletion)
+Mostly used workflows:
```
Comment (dpark01) [inline docs/workflows.rst:6]:
```suggestion
```

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 4]
```diff
@@ -6,14 +6,24 @@ task qiime_import_from_bam {
     }
     input { 
         Array[File] reads_bam
-        Int     memory_mb = 7000
+        Int    memory_mb = 7000
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:9]: Let's remove any edits from this PR that aren't strictly about documentation

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 5]
```diff
@@ -69,17 +79,45 @@ task trim_reads {
 
     input {
         File    reads_qza
-        #Boolean not_default = false
         String  forward_adapter         = "CTGCTGCCTCCCGTAGGAGT"
         String  reverse_adapter         = "AGAGTTTGATCCTGGCTCAG"
-        Int    min_length              = 1
+        Int     min_length              = 1
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:84]: Same as above -- even though it's nice cleanup, let's keep that to a different PR and keep this focused on documentation

File: pipes/WDL/tasks/tasks_assembly.wdl [hunk 6]
```diff
@@ -175,17 +277,33 @@ task ivar_trim {
...
       Int?   machine_mem_gb
       String docker = "andersenlabapps/ivar:1.3.1"
-    }
-
-    String  bam_basename=basename(aligned_bam, ".bam")
-    Int disk_size = 375
 
+      String  bam_basename=basename(aligned_bam, ".bam")
+      Int disk_size = 375
+    }
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_assembly.wdl:283]: this is an actual functional change -- let's take it out of this PR

File: pipes/WDL/tasks/tasks_assembly.wdl [hunk 7]
```diff
@@ -775,7 +957,7 @@ task filter_bad_ntc_batches {
 
         # re-index demux_meta lookup table by sample_original instead of sample_sanitized
         demux_meta = dict((v['sample_original'],v) for k,v in demux_meta_orig.items())
-
+
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_assembly.wdl:960]: let's remove this

File: pipes/WDL/workflows/assemble_refbased.wdl [hunk 8]
```diff
@@ -30,10 +31,6 @@ workflow assemble_refbased {
         aligner: {
             description: "Read aligner software to use. Options: novoalign, bwa, minimap2. Minimap2 can automatically handle Illumina, PacBio, or Oxford Nanopore reads as long as the 'PL' field in the BAM read group header is set properly (novoalign and bwa are Illumina-only)."
         }
-        novocraft_license: {
```
Comment (dpark01) [inline pipes/WDL/workflows/assemble_refbased.wdl:33]: While we don't use this much these days, this is still an optional parameter in the workflow and we should keep the documentation for it

**Review by dpark01 — CHANGES_REQUESTED** [review]
A few quick comments

**Comment by dpark01** [comment]
One quick note: I went into readthedocs and activated automatic public builds of the `fnegrete_test` branch so we can check there for how things render as you edit this PR.

See here for rendered output: https://viral-pipelines.readthedocs.io/en/fnegrete_test/

See here for build status (right now it's succeeding so nothing to see here, but if for some reason you had a syntax error or something, you'd see failures here): https://readthedocs.org/projects/viral-pipelines/builds/

File: pipes/WDL/workflows/16S_train_classifier.wdl (hunk 1 above)
Comment (golu099) [reply pipes/WDL/workflows/16S_train_classifier.wdl:1]: Deleted WDL

File: .vscode/settings.json (hunk 2 above)
Comment (golu099) [reply .vscode/settings.json]: Changed

File: pipes/WDL/tasks/tasks_assembly.wdl [hunk 9]
```diff
@@ -86,6 +126,67 @@ task scaffold {
...
+      scaffold_min_pct_contig_aligned: {
+        description: "Minimum percent of contig length in alignment (0.0 - 1.0)",
+        category: "advanced"
+      }
+      scaffold_fasta: {
+        description: "FASTA with best computed scaffolds from assembled read.",
+        patterns: ["*.fasta"],
+        category: "other"
+      }
+  #didnt include other outputs, not sure if i should
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_assembly.wdl:188]:
```suggestion
```

**Review by dpark01 — CHANGES_REQUESTED** [review]

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 10]
```diff
@@ -61,7 +61,7 @@ task qiime_import_from_bam {
     }
     runtime {
         docker: docker
-        memory: "~{memory_mb} MiB"
+        memory: "~{memory_mb} GB"
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:64]: Wait no, don't ask for 7000GB of RAM! If you change this to GB, 1) change the variable name from `memory_mb` to `memory_gb` and 2) change the default values down by a factor of 1000.
```suggestion
        memory: "~{memory_mb} MiB"
```

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 11]
```diff
@@ -142,7 +142,7 @@ task trim_reads {
 
     runtime {
         docker: docker
-        memory: "${memory_mb} MiB"
+        memory: "${memory_mb} GB"
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:145]:
```suggestion
        memory: "${memory_mb} MiB"
```

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 12]
```diff
@@ -192,7 +192,7 @@ task join_paired_ends {
     }
     runtime {
         docker: docker
-        memory: "${memory_mb} MiB"
+        memory: "${memory_mb} GB"
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:195]:
```suggestion
        memory: "${memory_mb} MiB"
```

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 13]
```diff
@@ -267,7 +267,7 @@ task deblur {
...
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:270]:
```suggestion
        memory: "${memory_mb} MiB"
```

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 14]
```diff
@@ -355,7 +355,7 @@ task train_classifier {
...
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:358]:
```suggestion
        memory: "${memory_mb} MiB"
```

**Review by dpark01 — CHANGES_REQUESTED** [review]

File: pipes/WDL/tasks/tasks_assembly.wdl [hunk 15]
```diff
@@ -86,6 +126,66 @@ task scaffold {
...
+      }
+
+      contigs_fasta: {
+        description: "De novo assembled fasta file",
+        patterns: ["*.fasta"],
+        category: "required"
+      }
+
+      reference_genome_fasta: {
+        description: "Reference genome to impute with.",
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_assembly.wdl:143]:
```suggestion
        description: "Reference genomes to scaffold against. Multiple reference genomes may be provided, one per fasta file, and this task will attempt each of them and select the reference that produces the most complete output. Each reference genome should be in a single fasta file with all segments/chromosomes contained in the file as separate sequences in the correct order. Output genomes from this task will contain the same number of sequences as the selected input reference genome, and will be emitted in the same order.",
```

File: pipes/WDL/tasks/tasks_assembly.wdl [hunk 16]
```diff
@@ -86,6 +126,66 @@ task scaffold {
...
+        category: "advanced"
+      }
+      nucmer_min_match: {
+        description: "Minimum length of an maximal exact match.",
+        category: "advanced"
+      }
+        nucmer_min_cluster:{
+        description: "Minimum cluster length (Default = 65 (nucmer), Default = 20 (promer)",
+        category: "advanced"
+      }
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_assembly.wdl:174]:
```suggestion
      nucmer_max_gap: {
        description: "When scaffolding contigs to the reference via nucmer, this specifies the -g parameter to nucmer (the maximum allowed gap between adjacent matches in a cluster). Our default is 200 (up from nucmer default of 90), mummer documentation suggests it is valid to increase up to 1000 to allow for more diversity.",
        category: "advanced"
      }
      nucmer_min_match: {
        description: "When scaffolding contigs to the reference via nucmer, this specifies the -l parameter to nucmer (the minimal size of a maximal exact match). Our default is 10 (down from nucmer default of 20) to allow for more divergence.",
        category: "advanced"
      }
        nucmer_min_cluster:{
        description: "When scaffolding contigs to the reference via nucmer, this specifies the -c parameter to nucmer (minimum cluster length). Our default is the nucmer default of 65 bp.",
        category: "advanced"
      }
```

**Review by dpark01 — CHANGES_REQUESTED** [review]

**Comment by golu099** [comment]
Did we merge this? Should I keep it open for future documentation fixes?

**Comment by golu099** [comment]
amplicon16S analysis wdl renamed to classify_qiime2_multi

Noticed that the old name is on the master branch still, so we should updated this

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 17]
```diff
@@ -69,17 +79,45 @@ task trim_reads {
...
+        forward_adapter: {
+            description: "Forward amplicon primer sequence.",
+            category: "advanced"
+        }
+        reverse_adapter: {
+            description: "Reverse amplicon primer sequence.",
+            cateogry: "advanced"
+        }
+        min_length: {
+            description: "Minimum length of the read, cutadapt will discard anything that is shorter than n bp AFTER trimming.Set to default.",
```
[the remark this answers is not in the document]
Comment (tomkinsc) [reply pipes/WDL/tasks/tasks_16S_amplicon.wdl:105]: After this small change, let's get this merged in.

**Review by tomkinsc — CHANGES_REQUESTED** [review]

**Review by tomkinsc — APPROVED** [review]
